moveit_ros/perception/pointcloud_octomap_updater/include/moveit/pointcloud_octomap_updater/pointcloud_octomap_updater.h

```diff
@@ -64,8 +72,29 @@ class PointCloudOctomapUpdater : public OccupancyMapUpdater
   void start() override;
   void stop() override;
   ShapeHandle excludeShape(const shapes::ShapeConstPtr& shape) override;
+
+  /** @brief Exclude a shape from the octomap when processing.  Allows specifying an initial transform to put in the
+     transform cache.
+             For use outside of the perception pipeline
+      @param shape The shape to exclude
+      @param pose Pose of the shape in the sensor frame */
+  ShapeHandle excludeShape(const shapes::ShapeConstPtr& shape, const Eigen::Isometry3d& pose);
+
   void forgetShape(ShapeHandle handle) override;
 
+  /** @brief process a pointcloud message and update the octomap.  Does not update the transform cache.
+             Can be used outside of the perception pipeline
+      @param cloud_msg The pointcloud to process
+      @param sensor_pose Eigen pose of the frame in which the pointcloud is given
+      @param update_method Whether to update the current octomap probabilities or to forget all previous octomap data */
+  bool processCloud(const sensor_msgs::PointCloud2::ConstPtr& cloud_msg, const Eigen::Isometry3d& sensor_pose,
+                    UpdateMethod update_method);
+
+  /** @brief process a pointcloud message and update the octomap.  Updates transform cache.
+             Requires setTransformCacheCallback to have been called
```

Comment:
The difference between these two methods doesn't become clear yet. Please augment the description.
```suggestion
  /** @brief Process a pointcloud message and update the octomap.
      Updates transform cache. Requires setTransformCacheCallback to have been called.
```